notebooks/07_emg_hand_gesture_classification.ipynb

```diff
@@ -117,7 +117,8 @@
    "metadata": {},
```

Comment:
<p>Code design: Just 2 complementary proposals, I let you decide if you feel this is important enough to dedicate your time to it.</p><p> </p><p>Proposal 1:</p><p>Would it make more sense to think about an ExpResults object as being linked to one single experiment, and called Experiment? Thus, with 4 experiments, you would instantiate 4 objects. This way, instead of having dictionaries of lists within a single big object, you have one list of multiple objects with attributes.</p><p> </p><p>The attributes of the object would be:</p><ul><li>self.name (name of the experiment), eg mg_*</li><li>self.ids or directly self.time_series?</li><li>self.pipeline?</li><li>self.session_type: either intra session or inter session, or maybe "self.train_session_id", "self.test_session_id"</li><li>self.subject_type: either intra subject or inter subject, or maybe "self.train_subject_id", "self.test_subject_id"</li><li>self.fit_time</li><li>self.score_time</li><li>self.test_score</li><li>self.train_score</li></ul><p>we could also add:</p><ul><li>self.statistical_method</li><li>self.geometric_method</li></ul><p>thinking forward about the possibility of comparing the impact of statistics versus geometry,</p><p>and individual methods:</p><ul><li>self.run(): would run cross_validate</li><li>self.print_result:</li><li>self.plot_result: would return an ax, so that plots of multiple experiments can be combined.</li></ul><p> </p><p>The method plot_results, would become a function, that either takes a list of ax as inputs, or a list of results.</p><p> </p><p>Proposal 2: Use MLFLOW</p><p> </p><p>Mlflow actually does exactly this. It is made to record experiments, their score, their attributes...and it provides multiple comparison tools right away. I am not sure that we want to add a dependency on mlflow to geomstats at this point, but you could install it in the notebook with:</p><p><code>!pip install mlflow</code> and use it. See:</p><p>https://www.mlflow.org/docs/latest/concepts.html</p><p>https://github.com/mlflow/mlflow/blob/master/examples/sklearn_logistic_regression/train.py</p><p> </p><p>If you want to log the results using mlflow, we could add a method:</p><ul><li>self.mlflow_log()</li></ul><p>that would log the different scores, metrics, and plots to mlflow.</p><p> </p><p>If you choose not to go with the object structure above nor mlflow, could you document what is expected as keys and values in the dictionary data structures, eg adding the Parameters section in the docstring for example. Otherwise it is hard to follow which keys are expected in which dictionary etc.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geomstats/geomstats/pull/912/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>